kernel/src/process/process_filter.rs

```diff
@@ -26,9 +26,14 @@ impl ProcessFilter {
             P_PID => Ok(ProcessFilter::WithPid(id)),
             P_PGID => Ok(ProcessFilter::WithPgid(id)),
             P_PIDFD => {
+                let fd = id.cast_signed();
+                if fd < 0 {
+                    return_errno_with_message!(Errno::EINVAL, "the pidfd is invalid");
+                }
+
```

Comment:
No need to have this blank line?

Reply:
Thanks for the throughout explanation. I'm in general agreement with your line of thought. This summarizes it well:
> We should group related things together to enhance the code readability.

My only counterpoint is that the way someone might arrange the code in blocks is tightly coupled to their particular thought process, which is a highly subjective experience.

Therefore, without an automated process (which I agree is hard) or very clear and explicit guidelines, adjusting these small details might be slightly conflicting (the writer's preference vs the reviewer's preference) or even counter-productive, depending on the extent to which such rules are relevant (which is also subjective!).

Anyway, just some food for thought 🙏🏼